connectors/sources/dropbox.py

```diff
@@ -844,11 +857,35 @@ async def close(self):
         await self.dropbox_client.close()
 
     async def ping(self):
-        endpoint = EndpointName.PING.value
-        if self._dls_enabled():
+        try:
             endpoint = EndpointName.AUTHENTICATED_ADMIN.value
-        await self.dropbox_client.ping(endpoint=endpoint)
-        self._logger.info("Successfully connected to Dropbox")
+            await self.dropbox_client.ping(endpoint=endpoint)
+            self._logger.info("Successfully connected to Dropbox")
+        except ClientResponseError as exception:
+            if str(exception.request_info.url) == AUTHENTICATED_ADMIN_URL:
+                endpoint = EndpointName.PING.value
+                await self.dropbox_client.ping(endpoint=endpoint)
+                self._logger.info("Successfully connected to Dropbox")
+            else:
+                raise
+
+    async def set_user_info(self):
+        try:
+            endpoint = EndpointName.AUTHENTICATED_ADMIN.value
+            response = await self.dropbox_client.ping(endpoint=endpoint)
+        except ClientResponseError as exception:
+            if str(exception.request_info.url) == AUTHENTICATED_ADMIN_URL:
+                endpoint = EndpointName.PING.value
+                response = await self.dropbox_client.ping(endpoint=endpoint)
+            else:
+                raise
+        _json = await response.json()
+        self.dropbox_client.member_id = _json.get("admin_profile", {}).get(
+            "team_member_id"
+        )
+        self.dropbox_client.root_namespace_id = _json.get("root_info", {}).get(
+            "root_namespace_id"
+        ) or _json.get("admin_profile", {}).get("root_folder_id")
```

Comment:
The following two dict calls can raise an raise an error:

1. `_json.get("admin_profile", {}).get("team_member_id")` if `{ admin_profile: None }`
2. `_json.get("admin_profile", {}).get("root_folder_id")` if `{ admin_profile: None }`
3. `_json.get("root_info", {}).get("root_namespace_id")` if `{ root_info: None }`

Can you null-safe these?

Also, if the json has `{ admin_profile: None }`, does that effectively mean Dropbox cannot sync documents? If so I think we need the following changes, otherwise the stack trace will not provide enough information to a user about what went wrong.

1. A detailed log explaining what is wrong with the response format
2. Raise a custom error (e.g. `DropboxResponseMalformed` error or something)